exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -41,26 +41,25 @@ def initialize(endpoint: config_opt('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', 'OTEL_
           raise ArgumentError, "unsupported compression key #{compression}" unless compression.nil? || compression == 'gzip'
           raise ArgumentError, 'headers must be comma-separated k=v pairs or a Hash' unless valid_headers?(headers)
 
-          uri = if endpoint == ENV['OTEL_EXPORTER_OTLP_ENDPOINT']
+          @uri = if endpoint == ENV['OTEL_EXPORTER_OTLP_ENDPOINT']
```

Comment:
To allow the trace_request method to be patched to allow network calls, we need a way to access the URI that the exporter has been configured with.